pygem/setup/config.py

```diff
@@ -7,42 +7,85 @@
 """
 import os
 import shutil
-import sys
 import yaml
+import argparse
+from ruamel.yaml import YAML
 
-# pygem_params file name
-config_fn = 'config.yaml'
-
-# Define the base directory and the path to the configuration file
-basedir = os.path.join(os.path.expanduser('~'), 'PyGEM')
-config_file = os.path.join(basedir, config_fn)  # Path where you want the config file
-
-# Get the source configuration file path from your package
-package_dir = os.path.dirname(__file__)  # Get the directory of the current script
-source_config_file = os.path.join(package_dir, config_fn)  # Path to params.py
-
-def ensure_config(overwrite=False):
-    isfile = os.path.isfile(config_file)
-    if isfile and overwrite:
-        overwrite = None
-        while overwrite is None:
-            # Ask the user for a y/n response
-            response = input(f"PyGEM configuration.yaml file already exist ({config_file}), do you wish to overwrite (y/n):").strip().lower()
-            # Check if the response is valid
-            if response == 'y' or response == 'yes':
-                overwrite = True
-            elif response == 'n' or response == 'no':
-                overwrite = False
-            else:
-                print("Invalid input. Please enter 'y' or 'n'.")
+class ConfigManager:
+    def __init__(self, config_filename='config.yaml', base_dir=None):
+        """initialize the ConfigManager class"""
+        self.config_filename = config_filename
+        self.base_dir = base_dir or os.path.join(os.path.expanduser('~'), 'PyGEM')
+        self.config_path = os.path.join(self.base_dir, self.config_filename)
+        self.source_config_path = os.path.join(os.path.abspath(__file__), "config.yaml")
```

Comment:
Whoops sorry, this should be
```suggestion
        self.source_config_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), "config.yaml")
```